### PR TITLE
Add Clock plugin

### PR DESCRIPTION
**Pull request by ms3056**
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/ms3056/clock

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [x]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

**Comment by ms3056** [comment]
Initial release.

**Comment by joethei** [comment]
Please remove the `main.js` file from the repo and add it to your .gitignore - compiled release files shouldn't get mixed with source code in your commits and should go to the releases files instead.
Same applies to the `data.json`

["authorUrl": "",](https://github.com/ms3056/clock/blob/5357b174f9a84e48a6ebff757a6d0560d2675982/manifest.json#L8)
If you don't have a website put your GitHub profile here.

["donation": "",](https://github.com/ms3056/clock/blob/5357b174f9a84e48a6ebff757a6d0560d2675982/manifest.json#L10)
Remove this.

[@import url("https://fonts.googleapis.com/css2?family=Roboto&display=swap");](https://github.com/ms3056/clock/blob/5357b174f9a84e48a6ebff757a6d0560d2675982/styles.css#LL1C1-L1C1)
Don't import font's through the web, read here how to do this instead: <https://docs.obsidian.md/Themes/App+themes/Embed+fonts+and+images+in+your+theme>

I also would not recommend against using a hardcoded font and hardcoded colors in general.
You can use the [CSS Variables](https://docs.obsidian.md/Reference/CSS+variables/CSS+variables) to style your plugin instead, this will make sure it will look good across many themes, without any tweaking.

[id: "clock-open",](https://github.com/ms3056/clock/blob/5357b174f9a84e48a6ebff757a6d0560d2675982/main.ts#L172)
No need to prefix this yourself, Obsidian will prefix the `id` with your plugin ID under the hood so this is redundant.

[name: "Open Clock",](https://github.com/ms3056/clock/blob/5357b174f9a84e48a6ebff757a6d0560d2675982/main.ts#LL173C4-L173C4)
Similarly, here, this will show up as `Clock: Open Clock` in the command pallete, so you can shorten this to just `clock`.

[if (this.app.workspace.layoutReady) {](https://github.com/ms3056/clock/blob/5357b174f9a84e48a6ebff757a6d0560d2675982/main.ts#L190-L200)
This should be implemented as follows:
```ts
this.app.workspace.onLayoutReady(() => {
	await this.initView();
});
```

[this.app.workspace.detachLeavesOfType(ClockViewType);](https://github.com/ms3056/clock/blob/5357b174f9a84e48a6ebff757a6d0560d2675982/main.ts#L208)
Do not detach leaves with your custom view in `onunload`, this is an antipattern, see: <https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Don't+detach+leaves+in+%60onunload%60>


[containerEl.createEl("h1", { text: "Clock Settings" });](https://github.com/ms3056/clock/blob/5357b174f9a84e48a6ebff757a6d0560d2675982/main.ts#L246)
[Only use headings under settings if you have more than one section. ](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)
[Avoid "settings" in settings headings](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Avoid+%22settings%22+in+settings+headings)


[// Only save the setting when 'Enter' is pressed](https://github.com/ms3056/clock/blob/5357b174f9a84e48a6ebff757a6d0560d2675982/main.ts#L280-L290)
Don't do this, this is not intuitive, also not fully mobile compatible.
Use [`onChange`](https://docs.obsidian.md/Plugins/User+interface/Settings#Register+a+settings+tab) instead.
